yarn.config.cjs

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+/*!
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/** @type {import('@yarnpkg/types')} */
+const { defineConfig } = require('@yarnpkg/types');
+const fg = require('fast-glob');
+
+/**
+ * The workspace object used in the constraints function
+ * @typedef {import('@yarnpkg/types').Yarn.Constraints.Workspace} Workspace
+ */
+
+module.exports = defineConfig({
+    async constraints({ Yarn }) {
+        /**
+         * Fetch a list of all the component workspaces using a glob pattern
+         * @type {string[]} components
+         */
+        const components = fg.sync('packages/*', {
```

Comment:
`fg.sync()` is called synchronously at the top level, which might block execution since there are a lot of packages.
can we use an async function and `await fg()`

Reply:
There's no difference here between using await or sync because the list of components must be present to validate against in the functions below. This is a _preference_ rather than an important requirement to update during code review. There is the same amount of "blocking execution" as you say, between awaiting an async function as there is using the synchronous version of a function. You only gain benefits from asynchronous if you are running tasks against it as the information becomes available, which we are _not_.